ml4gw/waveforms/sine_gaussian.py

```diff
@@ -0,0 +1,114 @@
+from dataclasses import dataclass
+
+import torch
+from scipy.signal.windows import tukey
+from torch import Tensor
+
+from ml4gw.types import ScalarTensor
+
+
+def semi_major_minor_from_e(e: Tensor):
+    a = 1.0 / torch.sqrt(2.0 - e * e)
+    b = a * torch.sqrt(1.0 - e * e)
+    return a, b
+
+
+# TODO: replace with torch implementation
+def tukey_window(num: int, alpha: float = 0.5):
+    return torch.tensor(tukey(num, alpha=alpha))
+
+
+@dataclass
+class SineGaussian:
+    sample_rate: float
+    duration: float
+
+    def __post_init__(self):
+        # determine times based on requested duration and sample rate
+        # and shift so that the waveform is centered at t=0
+        num = int(self.duration * self.sample_rate)
+        times = torch.arange(num, dtype=torch.float64) / self.sample_rate
+        times -= self.duration / 2.0
+
+        self.times = times
+        self.window = tukey_window(num)
```

Comment:
My general take is that if you don't necessarily need the dataclass attributes in the class itself, but just to initialize some other properties it depends on, it's probably better form to use a more traditional `__init__`. The main benefit of the `dataclass` is to avoid the boilerplate of `__init__`s that just assign their arguments to attributes and don't do much else. I'm a frequent abuser of this as well so not a big deal but this definitely feels like more than a container class for some data fields

Reply:
Yeah bad habit